### PR TITLE
Fix issue #398: add parse union at the end of cte

### PR DESCRIPTION
**Pull request by duongphannamhung**
Fix issue #398
add test with specific dialect (ansi), work well with Bigquery if we use "UNION ALL" rather than "UNION".

### CODE REVIEW

File: tests/test_others_dialect_specific.py
Comment (reata) [inline tests/test_others_dialect_specific.py:1]: If it's ANSI compatible, then don't put it into dialect specific test suits. `test_cte.py` is the better place for this test case.

File: tests/test_others_dialect_specific.py [hunk 1]
```diff
@@ -202,3 +202,30 @@ def test_merge_insert_row(dialect: str):
...
+FROM table2)
+SELECT
+table3.id, table3.name, table1.school
+FROM table3
+LEFT JOIN table1 ON table3.id = table1.id
+UNION
+SELECT
+table4.id, table4.name, table2.school
+FROM table4
+LEFT JOIN table2 ON table4.id = table2.id"""
```
Comment (reata) [inline tests/test_others_dialect_specific.py:225]: If you can simplify the test case, that would be even better. The corner case is with CTE + UNION. We just need one CTE instead of two. Even `INSERT INTO` can be removed.

**Review by reata — CHANGES_REQUESTED** [review]
Please make the change based on the comment inline. Ideally we want each test case to be as atomic as possible.
